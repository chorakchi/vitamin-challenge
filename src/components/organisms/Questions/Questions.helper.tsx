
  export const optionsProvider = (items : any) => ({
    chart: {
      type: "bubble",
      plotBorderWidth: 1,
      zoomType: "xy",
      height: (9 / 16) * 100 + "%", // 16:9 ratio
    },
    legend: {
      enabled: false,
    },
    title: {
      text: "",
    },

    subtitle: {
      text: "size of the point is absolute price change (24h)",
    },
    xAxis:{
      title: {
        text: 'Market Capitalization'
    },
  },
    yAxis:{
      title: {
        text: 'Volume (24h)'
    },
  },
    tooltip: {
      useHTML: true,
      headerFormat: "<table>",
      pointFormat:
        '<tr><th colspan="2"><h3>{point.name}</h3></th></tr>' +
        '<img src="https://s3-symbol-logo.tradingview.com/crypto/XTVC{point.symbol}.svg" width=52 height=52 />' +
        "<tr><th>Market Capitalization:</th><td>{point.x} $</td></tr>" +
        "<tr><th>Volume (24h):</th><td>{point.y} $</td></tr>" +
        "<tr><th>price change (24h):</th><td>{point.z}%</td></tr>",
      footerFormat: "</table>",
      followPointer: true,
    },

    plotOptions: {
      series: {
        dataLabels: {
          enabled: true,
          format: "{point.symbol}",
        },
      },
    },

    series: [
      {
        data: items,
      },
    ],
  })