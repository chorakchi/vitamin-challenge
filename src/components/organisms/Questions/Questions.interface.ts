import { HTMLAttributes } from "react";

type inputProps = {
  onChange: ({ target: any }) => void;
};

export interface QuestionsProps extends HTMLAttributes<HTMLElement> {
  nameProps?: inputProps;
  earnProps?: inputProps;
  riskProps?: inputProps;
}
