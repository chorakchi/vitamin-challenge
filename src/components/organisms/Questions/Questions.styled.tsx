import styled from "styled-components";
import { Box } from "components";

export const StyledWrapper = styled.div`
display: flex;
flex-direction: column ;
`;
