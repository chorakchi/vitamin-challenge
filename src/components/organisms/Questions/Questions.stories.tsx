import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Questions } from "./Questions";
export default {
  title: "Organisms/Questions",
  component: Questions,
} as ComponentMeta<typeof Questions>;

const Template: ComponentStory<typeof Questions> = (args) => (
  <Questions {...args} />
);

export const Questions_ = Template.bind({});
Questions_.args = {};
