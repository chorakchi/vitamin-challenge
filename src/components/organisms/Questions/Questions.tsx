import { StyledWrapper } from "./Questions.styled";
import { QuestionsProps } from "./Questions.interface";
import { TextField, Slider } from "components";

export const Questions: React.FC<QuestionsProps> = ({
  nameProps,
  earnProps,
  riskProps,
  ...props
}: QuestionsProps) => {
  return (
    <StyledWrapper {...props}>
      <TextField type="text" label="Whatâ€™s your name?" {...nameProps} />
      <TextField type="number" label="How much do you earn per month?" {...earnProps} />
      <Slider min="2" max="10" step="2" label="What is your risk tolerance level?" {...riskProps} />
    </StyledWrapper>
  );
};
