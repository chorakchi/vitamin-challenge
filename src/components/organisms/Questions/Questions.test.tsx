import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Questions } from "./Questions";

describe("Questions", () => {
  it("should render the Questions", () => {
    const subject = shallow(<Questions />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Questions />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
