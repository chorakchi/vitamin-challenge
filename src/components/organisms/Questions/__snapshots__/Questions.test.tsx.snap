// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions should render correctly and match to the Snapshot 1`] = `
<div
  className="Questionsstyled__StyledWrapper-sc-1yeop21-0 cNkGxQ"
>
  <div
    className="Stackstyled__StackStyled-sc-1adxq1u-0 bnrRyH"
    direction="column"
  >
    <label
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
    >
      Whatâ€™s your name?
      <br />
      <input
        className="TextFieldstyled__TextFieldStyled-sc-mvtmc2-0 NTHeK"
        type="text"
      />
    </label>
  </div>
  <div
    className="Stackstyled__StackStyled-sc-1adxq1u-0 bnrRyH"
    direction="column"
  >
    <label
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
    >
      How much do you earn per month?
      <br />
      <input
        className="TextFieldstyled__TextFieldStyled-sc-mvtmc2-0 NTHeK"
        type="number"
      />
    </label>
  </div>
  <span>
    <label
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
    >
      What is your risk tolerance level?
      <br />
      <input
        className="Sliderstyled__SliderStyled-sc-fj8kj3-0 ktJaBw"
        max="10"
        min="2"
        step="2"
        type="range"
      />
    </label>
  </span>
</div>
`;
